jupyterhub/app.py

```diff
@@ -1026,7 +1026,7 @@ def start_proxy(self):
         self.log.info("Starting proxy @ %s", self.proxy.public_server.bind_url)
         self.log.debug("Proxy cmd: %s", cmd)
         try:
-            self.proxy_process = Popen(cmd, env=env)
+            self.proxy_process = Popen(cmd, env=env, preexec_fn=os.setpgrp)
```

Comment:
@vilhelmen Thanks for the PR! I agree preexec is a reasonable approach. 

@minrk, I'm wondering if `os.setpgip` may be a more general solution that `os.setpgrp`. The POSIX standard has marked `setpgrp` as obsolete http://pubs.opengroup.org/onlinepubs/9699919799/